File: 04-WebGL/01-ProgrammablePipeline/22-RoboticArm/Stack.js
// functions use without 'this' prefix are private to class.
// functions with 'this' prefix are 'public' and hence when they use
function Stack()
{
    var elements=[];
    var top = 0;
   

    this.push=function(matrix)
    {
        var target = {};
        // code
        //elements[top] = matrix;
        elements[top] = Object.assign(target, matrix);
        top = top + 1;
    }
    
    this.pop=function()
    {
        top = top - 1;
        return elements.pop();
    }
   
}

File: 03-Android/01-ProgrammablePipeline/23-SolarSystem/app/src/main/java/com/example/solarSystem/stack.java
package com.example.solarSystem;

public class stack
{
	private int maxSize = 50;
	private float[][] data = new float[maxSize][16];
	private int top = -1;
	private float[] arr = new float[16];

	public void push(float[] matrix)
	{
		top++;
		for(int i=0; i<16; i++)
		{
			data[top][i] = matrix[i];
		}
		
	}

	public float[] pop()
	{
		for(int i=0; i<16; i++)
		{
			arr[i] = data[top][i];
		}
		top--;

		return (arr);
	}
}
